### PR TITLE
[TypeDeclaration] Add missing usage of ClassMethodReturnTypeOverrideGuard on ReturnTypeFromMockObjectRector

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

File: rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromMockObjectRector/Fixture/some_test_with_mock.php.inc [hunk 1]
```diff
@@ -8,7 +8,7 @@ use PHPUnit\Framework\TestCase;
 
 final class SomeTestWithMock extends TestCase
 {
-    public function createMock()
+    public function test()
```
Comment (samsonasik) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromMockObjectRector/Fixture/some_test_with_mock.php.inc:11]: I updated method name due to child without type is error early if parent method already have type, also consistent with example

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

File: rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromMockObjectRector/Fixture/some_test_with_mock.php.inc [hunk 2]
```diff
@@ -26,7 +26,7 @@ use PHPUnit\Framework\TestCase;
 
 final class SomeTestWithMock extends TestCase
 {
-    public function createMock(): \PHPUnit\Framework\MockObject\MockObject
+    public function test(): \PHPUnit\Framework\MockObject\MockObject
     {
         return $this->createMock('SomeType');
     }
```
Comment (TomasVotruba) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromMockObjectRector/Fixture/some_test_with_mock.php.inc:32]: Here the `createSomeMock()` should be used instead, as a factory method, not a test one.

Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromMockObjectRector/Fixture/some_test_with_mock.php.inc:32]: sure https://github.com/rectorphp/rector-src/pull/6171

**Pull request merged**